core/templates/pages/skill-editor-page/skill-preview-tab/skill-preview-tab.component.spec.ts

```diff
@@ -224,6 +235,27 @@ describe('Skill Preview Tab Component', () => {
     expect(component.displayCardIsInitialized).toEqual(true);
   });
 
+  it('should tell if current supplemental card is non empty', () => {
+    component.displayedCard = displayedCard;
+    spyOn(component, 'isSupplementalCardNonempty').and.returnValues(
+      true, false);
+
+    expect(component.isCurrentSupplementalCardNonEmpty()).toBeTrue();
+    expect(component.isCurrentSupplementalCardNonEmpty()).toBeFalse();
+  });
+
+  it('should tell if window can show two cards', () => {
+    spyOn(windowDimensionsService, 'getWidth').and.returnValue(
+      ExplorationPlayerConstants.TWO_CARD_THRESHOLD_PX + 1);
+
+    expect(component.canWindowShowTwoCards()).toBeTrue();
+  });
+
+  it('should tell if supplemental card is non empty', () => {
+    expect(component.isSupplementalCardNonempty(displayedCard))
+      .toBeFalse();
```

Comment:
Do we need a test for when this evaluates to true?

Reply:
Actually, we use the supplemental card only when the interaction in card is not inline that's why we expect it to be false only because when it is true we dont need the supplemetal card we just display the interaction with the card itself no need to extra block for it. So that's why we only expect it to be false onlly.

Reply:
Hm, but for the sake of consistency in the unit tests, can we have a test where we use a "displayedCard" that does not use an inline interaction and expect the output to be true?

Reply:
I have added condition to check for both inline and non inline by making a new dummy card which is supplemental/